### PR TITLE
[Bugfix][Connector-v2] fix file sink `isPartitionFieldWriteInFile` occurred exception when no columns are given

### PR DESCRIPTION
**Pull request by chovy-3012**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

***bug as below***:
I want to sink data from Kafka in text format, partitioned by date, into different HDFS directories, and the data does not contain timestamps.
test.conf :
```conf
env {
  job.name = "test"
  job.mode = "STREAMING"
  checkpoint.interval=20000
}

source {
  Kafka {
    result_table_name="test_table"
    bootstrap.servers = "xxxx:9092"
    topic = "test_topic"
    field_name = "data"
    format.type = "text"
    consumer.group = "test_group"
  }
}

transform {
  Sql {
      source_table_name = "test_table"
      result_table_name = "test_table_result"
      query="SELECT  data,formatdatetime(now(),'yyyy-MM-dd') new_ts   from  test_table  "
  }
}

sink {
   HdfsFile {
      fs.defaultFS = "hdfs://hadoop01:9000"
      is_enable_transaction = true
      have_partition = true
      partition_by = ["new_ts"]
      partition_dir_expression = "${v0}"
      is_partition_field_write_in_file =  false
      path = "/data/test"
      file_format_type = "json"
   }
}
```

cmd:
```shell
./bin/start-seatunnel-flink-13-connector-v2.sh test.conf
```

exception:
```shell
java.lang.UnsupportedOperationException
	at java.util.AbstractList.remove(AbstractList.java:161)
	at java.util.AbstractList$Itr.remove(AbstractList.java:374)
	at java.util.AbstractCollection.removeAll(AbstractCollection.java:376)
	at org.apache.seatunnel.connectors.seatunnel.file.sink.config.FileSinkConfig.<init>(FileSinkConfig.java:142)
	at org.apache.seatunnel.connectors.seatunnel.file.writer.FileSinkConfigTest.testInit(FileSinkConfigTest.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```

***bux fix***：

fix file sink `isPartitionFieldWriteInFile` occurred exception when no columns are given for the sink

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/config/FileSinkConfig.java [hunk 1]
```diff
@@ -83,7 +84,8 @@ public FileSinkConfig(@NonNull Config config, @NonNull SeaTunnelRowType seaTunne
 
         // if the config sink_columns is empty, all fields in SeaTunnelRowTypeInfo will being write
         if (CollectionUtils.isEmpty(this.sinkColumnList)) {
-            this.sinkColumnList = Arrays.asList(seaTunnelRowTypeInfo.getFieldNames());
+            this.sinkColumnList =
+                    new ArrayList<>(Arrays.asList(seaTunnelRowTypeInfo.getFieldNames()));
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/config/FileSinkConfig.java:90]: Could you add some comment in here. Only read the code will confused developers.

**Review by Hisoka-X — DISMISSED** [review]
LGTM expect one minor comment. cc @TyrantLucifer

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/config/FileSinkConfig.java (hunk 1 above)
Comment (chovy-3012) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/config/FileSinkConfig.java:90]: added comment in bug fix code

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
@chovy-3012 Could you retry the failed CI? https://github.com/chovy-3012/seatunnel/actions/runs/6492030625/job/17630262359

**Comment by chovy-3012** [comment]
> @chovy-3012 Could you retry the failed CI? https://github.com/chovy-3012/seatunnel/actions/runs/6492030625/job/17630262359

Retried more than 10 times , still failed.
Error messages show `no space left on device` , any suggestions?

error message as below:
```shell
[INFO] Running org.apache.seatunnel.connectors.seatunnel.jdbc.JdbcAutoGenerateSQLIT
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.34 s <<< FAILURE! - in org.apache.seatunnel.connectors.seatunnel.jdbc.JdbcAutoGenerateSQLIT
Error:  org.apache.seatunnel.connectors.seatunnel.jdbc.JdbcAutoGenerateSQLIT  Time elapsed: 8.34 s  <<< ERROR!
java.util.concurrent.CompletionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=postgres:14-alpine, imagePullPolicy=DefaultPullPolicy(), imageNameSubstitutor=org.testcontainers.utility.ImageNameSubstitutor$LogWrappedImageNameSubstitutor@489496a9)
Caused by: org.testcontainers.containers.ContainerFetchException: Failed to get Docker client for postgres:14-alpine
Caused by: com.github.dockerjava.api.exception.DockerClientException: Could not pull image: failed to register layer: write /usr/lib/libLLVM-15.so: no space left on device
```

**Review by hailin0 — APPROVED** [review]

**Pull request merged**